tests/components/wled/test_init.py

```diff
@@ -0,0 +1,76 @@
+"""Tests for the WLED integration."""
+import aiohttp
+from asynctest import patch
+import pytest
+
+from homeassistant.components.wled import async_setup_entry, async_unload_entry
+from homeassistant.components.wled.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_MAC, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from tests.common import MockConfigEntry, load_fixture
+from tests.components.wled import init_integration
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_config_entry_not_ready(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the WLED configuration entry not ready."""
+    aioclient_mock.get("http://example.local:80/json/", exc=aiohttp.ClientError)
+
+    entry = MockConfigEntry(
+        domain=DOMAIN, data={CONF_HOST: "example.local", CONF_MAC: "aabbccddeeff"}
+    )
+
+    with pytest.raises(ConfigEntryNotReady):
+        await async_setup_entry(hass, entry)
```

Comment:
See above.

Reply:
That won't work in this case, since the exception needs to be caught by the test.

Reply:
Instead check the state of the config entry:
https://github.com/home-assistant/home-assistant/blob/ac4d8ee07f8164571b5223c90207e84b5a75efbd/homeassistant/config_entries.py#L200-L201

Reply:
Ok, check, will do.
One small notice; this is not the way how it is usually done in our codebase. Nevertheless, I see your point.

Reply:
Adjusted, much cleaner, thank you!